module/zfs/zfs_ioctl.c

```diff
@@ -1122,6 +1123,12 @@ zfs_secpolicy_create_clone(zfs_cmd_t *zc, nvlist_t *innvl, cred_t *cr)
 	    ZFS_DELEG_PERM_CREATE, cr)) != 0)
 		return (error);
 
+	/* Mount permission is not required for a ZVOL */
+	if (nvlist_lookup_int32(innvl, "type", &type) != 0)
+		return (SET_ERROR(EINVAL));
+	if (type == DMU_OST_ZVOL)
+		return (0);
+
 	return (zfs_secpolicy_write_perms(parentname,
 	    ZFS_DELEG_PERM_MOUNT, cr));
```

Comment:
Then why don't we rework this a little bit so mounting only applies to filesystems.  How about this?

```c
        /* Mount permission is only requires for filesystems */
	error = nvlist_lookup_int32(innvl, "type", &type);
	if (error != 0) {
                return (SET_ERROR(EINVAL));
        } else if (type == DMU_OST_ZFS) {
        ￼  	error = zfs_secpolicy_write_perms(parentname,
￼  	            ZFS_DELEG_PERM_MOUNT, cr));
        }

        return (error);
```